components/motor/ulnstepper/28byj-48.go

```diff
@@ -0,0 +1,372 @@
+// Package unipolarfivewirestepper implements a GPIO based
+// stepper motor (model: 28byj-48) with uln2003 controler.
+package unipolarfivewirestepper
+
+/*
+	Motor Name:  		28byj-48
+	Motor Controler: 	ULN2003
+	Datasheet:
+			ULN2003: 	https://www.makerguides.com/wp-content/uploads/2019/04/ULN2003-Datasheet.pdf
+			28byj-48:	https://components101.com/sites/default/files/component_datasheet/28byj48-step-motor-datasheet.pdf
+
+	This driver will drive the motor with half-step driving method (instead of full-step drive) for higher resolutions.
+	In half-step the current vector divides a circle into eight parts. The eight step switching sequence is shown in
+	stepSequence below. The motor takes 5.625*(1/64)° per step. For 360° the motor will take 4096 steps.
+*/
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var (
+	model                = resource.NewDefaultModel("28byj48")
+	minDelayBetweenTicks = 100 * time.Microsecond // minimum sleep time between each ticks
```

Comment:
this translates to ~146rpm. if you want to enforce and document a lower number, do that.
10000 / 4096 * 60.